Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -298,11 +297,11 @@ - (FSTFieldValue *)parseArray:(NSArray *)array context:(ParseContext &&)context
 - (void)parseSentinelFieldValue:(FIRFieldValue *)fieldValue context:(ParseContext &&)context {
   // Sentinels are only supported with writes, and not within arrays.
   if (!context.write()) {
-    FSTThrowInvalidArgument(@"%@ can only be used with updateData() and setData()%s",
-                            fieldValue.methodName, context.FieldDescription().c_str());
+    ThrowInvalidArgument("%s can only be used with updateData() and setData()%s",
```

Comment:
Nit: should there be a space in-between `setData()` and `%s`? I'm not sure how the resulting sentence may look like, and I see this pattern used quite a lot in this file, so perhaps I just don't understand the intended effect.

Reply:
`FieldDescription` is either the empty string or contains its own leading space:

```c++
std::string ParseContext::FieldDescription() const {
  // TODO(b/34871131): Remove nullptr check once we have proper paths for fields
  // within arrays.
  if (!path_ || path_->empty()) {
    return "";
  } else {
    return util::StringFormat(" (found in field %s)", path_->CanonicalString());
  }
}
```
This is correct.